data/movements/scripts/field.lua

```diff
@@ -0,0 +1,57 @@
+local fields = {
+	[1423] = {type = CONDITION_FIRE, list = DAMAGELIST_VARYING_PERIOD, stepDamage = 20, periodicDamage = 10, period = {10, 12}, rounds = 2}, -- campfire
```

Comment:
This is just a minor thing but you could probably think about structuring your table different as you re use same stuff quite often maybe something like:
```lua
local fields = {
	[{1497, 1498, 1499, 11098, 11099}] = {harmless = true},
	[{1423, 1424, 1425, 1487}] = {type = CONDITION_FIRE, list = DAMAGELIST_VARYING_PERIOD, stepDamage = 20, periodicDamage = 10, period = {10, 12}, rounds = 2},
	[{1492, 1500, 10986}] = {type = CONDITION_FIRE, list = DAMAGELIST_VARYING_PERIOD, stepDamage = 20, periodicDamage = 10, period = {10, 12}, rounds = 7},
	[{1493, 1501, 10987}] = {type = CONDITION_FIRE, list = DAMAGELIST_VARYING_PERIOD, stepDamage = 10, periodicDamage = 10, period = {10, 12}, rounds = 7},
	[{1506, 1507, 7465, 7466, 7467, 7468, 7469, 7470, 7471, 7472, 7473}] = {type = CONDITION_FIRE, stepDamage = 300},
	[{1490, 1496, 1503}] = {type = CONDITION_POISON, list = DAMAGELIST_LOGARITHMIC_DAMAGE, stepDamage = 5, periodicDamage = 5},
	[{1491, 1495, 1504}] = {type = CONDITION_ENERGY, list = DAMAGELIST_VARYING_PERIOD, stepDamage = 30, periodicDamage = 25, period = {8, 10}},
	[{10988}] = {type = CONDITION_FIRE, list = DAMAGELIST_VARYING_PERIOD, stepDamage = 10, periodicDamage = 5, period = {10, 12}, rounds = 7}
}
```

Reply:
This makes up for simplifying the amount of tables with exact the same types of data within the main table ``fields`` but does not adds for customization, but for complexity, I'd say the simplest solution is the best solution.

Reply:
Well I don't think it adds to much complexety to it, you can still index with normal table index if it's a single table index
```lua
local fields = {
	[{1497, 1498, 1499, 11098, 11099}] = {harmless = true},
	[{1423, 1424, 1425, 1487}] = {type = CONDITION_FIRE, list = DAMAGELIST_VARYING_PERIOD, stepDamage = 20, periodicDamage = 10, period = {10, 12}, rounds = 2},
	[{1492, 1500, 10986}] = {type = CONDITION_FIRE, list = DAMAGELIST_VARYING_PERIOD, stepDamage = 20, periodicDamage = 10, period = {10, 12}, rounds = 7},
	[{1493, 1501, 10987}] = {type = CONDITION_FIRE, list = DAMAGELIST_VARYING_PERIOD, stepDamage = 10, periodicDamage = 10, period = {10, 12}, rounds = 7},
	[{1506, 1507, 7465, 7466, 7467, 7468, 7469, 7470, 7471, 7472, 7473}] = {type = CONDITION_FIRE, stepDamage = 300},
	[{1490, 1496, 1503}] = {type = CONDITION_POISON, list = DAMAGELIST_LOGARITHMIC_DAMAGE, stepDamage = 5, periodicDamage = 5},
	[{1491, 1495, 1504}] = {type = CONDITION_ENERGY, list = DAMAGELIST_VARYING_PERIOD, stepDamage = 30, periodicDamage = 25, period = {8, 10}},
	[10988] = {type = CONDITION_FIRE, list = DAMAGELIST_VARYING_PERIOD, stepDamage = 10, periodicDamage = 5, period = {10, 12}, rounds = 7}
}

function onStepIn(creature, item, position, fromPosition)
	local field = fields[item:getId()]
	if not field then
		for k,v in pairs(fields) do
			if isInArray(k, item:getId()) then
				field = v
			end
		end
	end
	if field.harmless or item:getType():isBlocking() then
		if not creature:isInGhostMode() then
			item:remove()
		end
		return true
	end

	Creature.addDamageCondition(item:getOwner(), creature, field.type, field.list, {step = field.stepDamage, periodic = field.periodicDamage}, field.period, field.rounds, item)
	return true
end

function onAddItem(item, tile, position)
	local owner = item:getOwner()
	local field = fields[item:getId()]
	if not field then
		for k,v in pairs(fields) do
			if isInArray(k, item:getId()) then
				field = v
			end
		end
	end
	for _, target in ipairs(Tile(position):getCreatures()) do
		Creature.addDamageCondition(owner, target, field.type, field.list, {step = field.stepDamage, periodic = field.periodicDamage}, field.period, field.rounds, item)
	end
	return true
end
```

Reply:
Well I don't think it adds to much complexety to it, you can still index with normal table index if it's a single table index
```lua
local fields = {
	[{1497, 1498, 1499, 11098, 11099}] = {harmless = true},
	[{1423, 1424, 1425, 1487}] = {type = CONDITION_FIRE, list = DAMAGELIST_VARYING_PERIOD, stepDamage = 20, periodicDamage = 10, period = {10, 12}, rounds = 2},
	[{1492, 1500, 10986}] = {type = CONDITION_FIRE, list = DAMAGELIST_VARYING_PERIOD, stepDamage = 20, periodicDamage = 10, period = {10, 12}, rounds = 7},
	[{1493, 1501, 10987}] = {type = CONDITION_FIRE, list = DAMAGELIST_VARYING_PERIOD, stepDamage = 10, periodicDamage = 10, period = {10, 12}, rounds = 7},
	[{1506, 1507, 7465, 7466, 7467, 7468, 7469, 7470, 7471, 7472, 7473}] = {type = CONDITION_FIRE, stepDamage = 300},
	[{1490, 1496, 1503}] = {type = CONDITION_POISON, list = DAMAGELIST_LOGARITHMIC_DAMAGE, stepDamage = 5, periodicDamage = 5},
	[{1491, 1495, 1504}] = {type = CONDITION_ENERGY, list = DAMAGELIST_VARYING_PERIOD, stepDamage = 30, periodicDamage = 25, period = {8, 10}},
	[10988] = {type = CONDITION_FIRE, list = DAMAGELIST_VARYING_PERIOD, stepDamage = 10, periodicDamage = 5, period = {10, 12}, rounds = 7}
}

function onStepIn(creature, item, position, fromPosition)
	local field = fields[item:getId()]
	if not field then
		for k,v in pairs(fields) do
			if isInArray(k, item:getId()) then
				field = v
			end
		end
	end
	if field.harmless or item:getType():isBlocking() then
		if not creature:isInGhostMode() then
			item:remove()
		end
		return true
	end

	Creature.addDamageCondition(item:getOwner(), creature, field.type, field.list, {step = field.stepDamage, periodic = field.periodicDamage}, field.period, field.rounds, item)
	return true
end

function onAddItem(item, tile, position)
	local owner = item:getOwner()
	local field = fields[item:getId()]
	if not field then
		for k,v in pairs(fields) do
			if isInArray(k, item:getId()) then
				field = v
			end
		end
	end
	for _, target in ipairs(Tile(position):getCreatures()) do
		Creature.addDamageCondition(owner, target, field.type, field.list, {step = field.stepDamage, periodic = field.periodicDamage}, field.period, field.rounds, item)
	end
	return true
end
```

Reply:
Well I don't think it adds to much complexety to it, you can still index with normal table index if it's a single table index
```lua
local fields = {
	[{1497, 1498, 1499, 11098, 11099}] = {harmless = true},
	[{1423, 1424, 1425, 1487}] = {type = CONDITION_FIRE, list = DAMAGELIST_VARYING_PERIOD, stepDamage = 20, periodicDamage = 10, period = {10, 12}, rounds = 2},
	[{1492, 1500, 10986}] = {type = CONDITION_FIRE, list = DAMAGELIST_VARYING_PERIOD, stepDamage = 20, periodicDamage = 10, period = {10, 12}, rounds = 7},
	[{1493, 1501, 10987}] = {type = CONDITION_FIRE, list = DAMAGELIST_VARYING_PERIOD, stepDamage = 10, periodicDamage = 10, period = {10, 12}, rounds = 7},
	[{1506, 1507, 7465, 7466, 7467, 7468, 7469, 7470, 7471, 7472, 7473}] = {type = CONDITION_FIRE, stepDamage = 300},
	[{1490, 1496, 1503}] = {type = CONDITION_POISON, list = DAMAGELIST_LOGARITHMIC_DAMAGE, stepDamage = 5, periodicDamage = 5},
	[{1491, 1495, 1504}] = {type = CONDITION_ENERGY, list = DAMAGELIST_VARYING_PERIOD, stepDamage = 30, periodicDamage = 25, period = {8, 10}},
	[10988] = {type = CONDITION_FIRE, list = DAMAGELIST_VARYING_PERIOD, stepDamage = 10, periodicDamage = 5, period = {10, 12}, rounds = 7}
}

function onStepIn(creature, item, position, fromPosition)
	local field = fields[item:getId()]
	if not field then
		for k,v in pairs(fields) do
			if isInArray(k, item:getId()) then
				field = v
			end
		end
	end
	if field.harmless or item:getType():isBlocking() then
		if not creature:isInGhostMode() then
			item:remove()
		end
		return true
	end

	Creature.addDamageCondition(item:getOwner(), creature, field.type, field.list, {step = field.stepDamage, periodic = field.periodicDamage}, field.period, field.rounds, item)
	return true
end

function onAddItem(item, tile, position)
	local owner = item:getOwner()
	local field = fields[item:getId()]
	if not field then
		for k,v in pairs(fields) do
			if isInArray(k, item:getId()) then
				field = v
			end
		end
	end
	for _, target in ipairs(Tile(position):getCreatures()) do
		Creature.addDamageCondition(owner, target, field.type, field.list, {step = field.stepDamage, periodic = field.periodicDamage}, field.period, field.rounds, item)
	end
	return true
end
```

Reply:
Well I don't think it adds to much complexety to it, you can still index with normal table index if it's a single table index
```lua
local fields = {
	[{1497, 1498, 1499, 11098, 11099}] = {harmless = true},
	[{1423, 1424, 1425, 1487}] = {type = CONDITION_FIRE, list = DAMAGELIST_VARYING_PERIOD, stepDamage = 20, periodicDamage = 10, period = {10, 12}, rounds = 2},
	[{1492, 1500, 10986}] = {type = CONDITION_FIRE, list = DAMAGELIST_VARYING_PERIOD, stepDamage = 20, periodicDamage = 10, period = {10, 12}, rounds = 7},
	[{1493, 1501, 10987}] = {type = CONDITION_FIRE, list = DAMAGELIST_VARYING_PERIOD, stepDamage = 10, periodicDamage = 10, period = {10, 12}, rounds = 7},
	[{1506, 1507, 7465, 7466, 7467, 7468, 7469, 7470, 7471, 7472, 7473}] = {type = CONDITION_FIRE, stepDamage = 300},
	[{1490, 1496, 1503}] = {type = CONDITION_POISON, list = DAMAGELIST_LOGARITHMIC_DAMAGE, stepDamage = 5, periodicDamage = 5},
	[{1491, 1495, 1504}] = {type = CONDITION_ENERGY, list = DAMAGELIST_VARYING_PERIOD, stepDamage = 30, periodicDamage = 25, period = {8, 10}},
	[10988] = {type = CONDITION_FIRE, list = DAMAGELIST_VARYING_PERIOD, stepDamage = 10, periodicDamage = 5, period = {10, 12}, rounds = 7}
}

function onStepIn(creature, item, position, fromPosition)
	local field = fields[item:getId()]
	if not field then
		for k,v in pairs(fields) do
			if isInArray(k, item:getId()) then
				field = v
			end
		end
	end
	if field.harmless or item:getType():isBlocking() then
		if not creature:isInGhostMode() then
			item:remove()
		end
		return true
	end

	Creature.addDamageCondition(item:getOwner(), creature, field.type, field.list, {step = field.stepDamage, periodic = field.periodicDamage}, field.period, field.rounds, item)
	return true
end

function onAddItem(item, tile, position)
	local owner = item:getOwner()
	local field = fields[item:getId()]
	if not field then
		for k,v in pairs(fields) do
			if isInArray(k, item:getId()) then
				field = v
			end
		end
	end
	for _, target in ipairs(Tile(position):getCreatures()) do
		Creature.addDamageCondition(owner, target, field.type, field.list, {step = field.stepDamage, periodic = field.periodicDamage}, field.period, field.rounds, item)
	end
	return true
end
```

Reply:
As you can see in your solution, using tables as tables indexes to reduce the amout of tables with the exact same data will require just a bunch of not needed code, and that also raises an issue where one would not be able to individually customize a single table for a specific item id, and that's what we have to have in mind when creating such scripts, giving the simplest solution and most efficient solution, so that each user can customize it for their own needs (like you showed in the example), if needed.

Reply:
> 
> 
> that also raises an issue where one would not be able to individually customize a single table for a specific item id

Nope that's actually not the case, you can still customize as much as you want easily, in fact
I'd even say it's easier, as you can group new fields and such into the already created index or just create a new one for it. Which will not result in a table mess later on.
```lua
if not field then
	for k,v in pairs(fields) do
		if type(k) ~= "table" then
			goto continue
		end
		if isInArray(k, item:getId()) then
			field = v
		end
		::continue::
	end
end
``` 
this entire part will be skipped if the itemid is registered as a non table index, it's just a few extra lines of code reading for more efficiency when it comes up to table reading

Reply:
I meant that you would not be able to customize it on the ``local`` table ``fields`` (when loading script - all data would be there, for the user to see and configure individually.), but rather on run-time, which would obfuscate its data - and require more caution, and hence more code - when editing the script whenever the user needed to perform such task.

But in any case, I'm open for compromises cause you clearly possess more knowledge than I do, and I'm also not a programmer (lack of effort), so we just need someone to review the current code and your proposal, and if approved a maintainer himself can edit the script as you suggested, if I don't do it first.